seqr/views/utils/dataset_utils.py

```diff
@@ -407,15 +421,15 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
         samples = validate_samples(samples, sample_id_to_tissue_type, warnings)
 
     # Delete old data
-    individual_db_ids = {s.individual_id for s in samples}
-    to_delete = model_cls.objects.filter(sample__individual_id__in=individual_db_ids).exclude(sample__data_source=data_source)
+    to_delete = model_cls.objects.filter(sample__in=samples).exclude(sample__data_source=data_source)
     prev_loaded_individual_ids = set(to_delete.values_list('sample__individual_id', flat=True))
     if to_delete:
         model_cls.bulk_delete(user, to_delete)
 
     loaded_sample_ids = set(model_cls.objects.filter(sample__in=samples).values_list('sample_id', flat=True).distinct())
+    samples = Sample.objects.select_related('individual__family__project').filter(id__in={s.id for s in samples} - loaded_sample_ids)
```

Comment:
I don't think we want to do a full new query for these- you already call `select_related` above so theres no need to call it again

Reply:
The new samples don't have `select_related` called. But I'm unsure about the performance impact if we don't do that for the new samples.

Reply:
I'm going to eliminate the new query. If the performance is an issue for new samples, we may add `selete_related` just for new sample models.